include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -173,31 +175,31 @@ class alignment_algorithm :
      * |space (begin positions) |\f$ O(n*m) \f$    |\f$ O(n*k) \f$     |
      * |space (alignment)       |\f$ O(n*m) \f$    |\f$ O(n*k) \f$     |
      */
-    template <indexed_sequence_pair_range indexed_sequence_pairs_t>
+    template <indexed_sequence_pair_range indexed_sequence_pairs_t,
+              typename callback_t>
     //!\cond
-        requires !traits_t::is_vectorised
+        requires !traits_t::is_vectorised &&
+                 is_type_specialisation_of_v<
+                    typename function_traits<std::remove_reference_t<callback_t>>::template argument_type_at<0>,
+                    alignment_result>
     //!\endcond
-    auto operator()(indexed_sequence_pairs_t && indexed_sequence_pairs)
+    void operator()(indexed_sequence_pairs_t && indexed_sequence_pairs, callback_t && callback)
     {
-        using sequence_pairs_t = std::tuple_element_t<0, std::ranges::range_value_t<indexed_sequence_pairs_t>>;
-        using sequence1_t = std::tuple_element_t<0, sequence_pairs_t>;
-        using sequence2_t = std::tuple_element_t<1, sequence_pairs_t>;
-        using result_t = typename align_result_selector<sequence1_t, sequence2_t, config_t>::type;
-
         using std::get;
 
-        std::vector<alignment_result<result_t>> results{};
         for (auto && [sequence_pair, idx] : indexed_sequence_pairs)
-            results.emplace_back(compute_single_pair(idx, get<0>(sequence_pair), get<1>(sequence_pair)));
-
-        return results;
+            compute_single_pair(idx, get<0>(sequence_pair), get<1>(sequence_pair), callback);
     }
 
     //!\overload
-    template <indexed_sequence_pair_range indexed_sequence_pairs_t>
-    auto operator()(indexed_sequence_pairs_t && indexed_sequence_pairs)
+    template <indexed_sequence_pair_range indexed_sequence_pairs_t, typename callback_t>
+    void operator()(indexed_sequence_pairs_t && indexed_sequence_pairs,
+                    callback_t && callback)
```

Comment:
put on line above

Reply:
Like above. I'd like to do it in the next PRs since I am changing this anyway.